### PR TITLE
Transfer: implement sending to multiple recipients 

### PR DESCRIPTION
**Pull request by xiphon**
Design is inspired by https://github.com/monero-project/monero-gui/issues/740 discussion. Thanks to all the participants.

Closes https://github.com/monero-project/monero-gui/issues/740.

Requires https://github.com/monero-project/monero-gui/pull/3325.

Only the last commit is a subject for review.

monero-send-to-multiple-recipients

### CODE REVIEW

**Comment by rating89us** [comment]
Very nice. Some feedback:
1. Fee estimative should receive a "Fee" title in white text (the same used in Fee priority). Fee value should be aligned below total amount.

image

2. Both inline buttons should stay inside the address field, and should be displayed only when it's empty (to save space for the address).

image

3. I miss fiat estimation of each destination, even though it is displayed in send confirmation dialog). But I understand we lack space here. Maybe we could list all amounts at the end (see below). Example:
```
Total 
         0.047 XMR
         (0.01 + 0.03 + 0.007 XMR)
         7.22 USD
         (1.54 + 4.61 + 1.07 USD)
```
If there are more than 3, just break the line

```
Total 
         0.077 XMR
         (0.01 + 0.03 + 0.007
         0.01 + 0.01 + 0.01 XMR)
         11.84 USD
         (1.54 + 4.61 + 1.07
          1.54 + 1.54 + 1.54 USD)
```

**Comment by selsta** [comment]
> I miss fiat estimation of each destination, even though it is displayed in send confirmation dialog. 

I think that would be a bit overkill, there is already a lot going on here with multiple recipients.

File: pages/Transfer.qml [hunk 1]
```diff
@@ -167,184 +163,448 @@ Rectangle {
...
+                                }
+                                validator: RegExpValidator {
+                                    regExp: /^\s*(\d{1,8})?([\.,]\d{1,12})?\s*$/
+                                }
+                            }
+
+                            MoneroComponents.TextPlain {
+                                Layout.preferredWidth: recipientLayout.thirdRowWidth
+                                horizontalAlignment: Text.AlignHCenter
+                                font.family: MoneroComponents.Style.fontRegular.name
```
Comment (selsta) [inline pages/Transfer.qml:424]:
```suggestion
                                font.family: FontAwesome.fontFamilySolid
                                font.styleName: "Solid"
```

**Review by selsta — COMMENTED** [review]
I did not get OpenAlias to work yet.

**Comment by GBKS** [comment]
Looks really good.

The inline scrolling of recipients in the modal is not that great. Maybe the default could be to just red "3 recipients", and clicking that would expand the full list (which would then also expand the modal)? That would keep the initial modal size small and still allow users to more conveniently review every single address without the cramped scroll interaction. Just an idea.

**Comment by xiphon** [comment]
@GBKS 

> Maybe the default could be to just red "3 recipients"

Current implementation displays 2 recipients with no scrolling.

> clicking that would expand the full list (which would then also expand the modal)?

It is ~1000px height list in 16 recipients case. I'm concerned it will look good.

File: pages/Transfer.qml (hunk 1 above)
Comment (xiphon) [reply pages/Transfer.qml]: Fixed

**Comment by xiphon** [comment]
> I did not get OpenAlias to work yet.

Fixed.

**Comment by GBKS** [comment]
> It is ~1000px height list in 16 recipients case. I'm concerned it will look good.

It is a tricky one. Easy to come up with some overly complex solutions for this, but let's just stick with your approach then.

**Comment by xiphon** [comment]
>     1. Fee estimative should receive a "Fee" title in white text and should be aligned below total amount.

Will give it a try in a while.

**Comment by xiphon** [comment]
Updated estimated fee layout.

monero-gui-send-multiple-recipients-fee

**Comment by xiphon** [comment]
>     1. Both inline buttons should stay inside the address field, and should be displayed only when it's empty (to save space for the address).

Considered this option as well. I don't like hiding some controls based on the user's input. IMO the current implementation looks better.

I'd like to hear other guys opinions on this.

**Comment by rating89us** [comment]
image
Good, but Fee lines have too much vertical spacing

**Comment by rating89us** [comment]
> IMO the current implementation looks better.

I always like to think about the function of each button, imagining a label explaining its action. Currently it's not easy for a user to know what these two buttons are going to do.

Currently it seems that these two buttons have two functions:
- "Add an address" (it adds an address when last field is empty)
- "Add a new recipient" (it creates a new recipient when the last field is not empty)

When inside the address field, both have a single function:
- "Add an address to this field" (it adds an address when this field is empty)

**Comment by xiphon** [comment]
> Good, but Fee lines have too much vertical spacing

image

**Comment by selsta** [comment]
Needs rebase.

**Comment by xiphon** [comment]
Rebased

**Comment by selsta** [comment]
> https://user-images.githubusercontent.com/26060097/107953419-046c1580-6f93-11eb-8ef1-6d4347c2d533.png

Minor comment, while it is technically correctly aligned with the text on the left side, it looks wrong in my eyes, especially on a large window. @GBKS any opinions on spacing here?

**Comment by GBKS** [comment]
Not an easy layout issue. The fee and total texts are all a bit floaty. How about bringing in a little more structure with these containers (see below)? That way there's a clear relationship what context each bit of information belongs to. Not totally clean, TBH.

monero-send-multiple-recipients-gbks-210323

**Comment by xiphon** [comment]
Looks like we're risking to end up in endless discussion about minor and not that important PR part. 

Reverted all fee layout changes i made during the review back to the original state.

Once we merge the PR, anyone is free to experiment with fee layout, open the discussion on topic, propose the changes or even get 'em implemented in a separate PR later.

**Review by selsta — APPROVED** [review]
ok, any further design changes can be done later

**Comment by selsta** [comment]
> How about bringing in a little more structure with these containers (see below)?

If we add fee again later it probably should be done this way, looks cleaner.

**Pull request merged**